extension/src/lib/wsConnector.js

```diff
@@ -0,0 +1,115 @@
+/* eslint-env node */
+/* global fluid, JSON */
+
+"use strict";
+
+var WebSocket = WebSocket || require("ws");
+var gpii = fluid.registerNamespace("gpii");
+
+fluid.defaults("gpii.wsConnector", {
+    gradeNames: "fluid.component",
+    solutionId: null,
+    flowManager: null,
+    retryTime: 5,  // Try reconnecting after x seconds.
+    members: {
+        socket: null
+    },
+    invokers: {
+        connect: {
+            funcName: "gpii.wsConnector.connect",
+            args: "{that}"
+        },
+        onopen: {
+            funcName: "gpii.wsConnector.onopen",
+            args: "{that}"
+        },
+        onerror: {
+            funcName: "gpii.wsConnector.onerror",
+            args: ["{that}",  "{arguments}.0"]
+        },
+        onclose: {
+            funcName: "gpii.wsConnector.onclose",
+            args: ["{that}",  "{arguments}.0"]
+        },
+        onmessage: {
+            funcName: "gpii.wsConnector.onmessage",
+            args: ["{that}",  "{arguments}.0"]
+        }
+    },
+    events: {
+        onCreate: null,
+        onConnect: null,
+        onConnectionSucceeded: null,
+        onErrors: null,
+        onMessage: null,
+        onSettingsChange: null
+    },
+    listeners: {
+        onCreate: "{wsConnector}.connect",
+        onConnect: {
+            funcName: "gpii.wsConnector.setup",
+            args: "{that}"
+        },
+        onErrors: {
+            funcName: "gpii.wsConnector.error",
+            args: ["{that}", "{arguments}.0", "{arguments}.1"]
+        }
+    }
+});
+
+gpii.wsConnector.onopen = function (that) {
```

Comment:
All these invokers are unnecessary - they can just be written out inline in the "listeners" block